### ISSUE TITLE
JRuby: Stacktraces from Java exceptions do not get sent

### ISSUE THREAD

**Issue by marshalium** [getsentry/raven-ruby#333]
I have found that when running a JRuby application, Java exceptions that are sent to Sentry are missing their stack traces. While Ruby exceptions don't have that problem.

It is getting stripped out somewhere Raven::Event. stacktrace_interface_from. I think it is trying to find the actual code to include context in the event, but it can't find it for Java code. I would prefer that it just fallback to the exact lines from e.backtrace when no context can be found.

Here is example code that can be used to reproduce the issue:
```
require 'ap'
require 'sentry-raven'

def raises_java_exception
  raise java.lang.OutOfMemoryError.new("A Java error")
end

def raises_ruby_exception
  raise RuntimeError.new("A Ruby error")
end

begin
  raises_java_exception
rescue Exception => e
  result = Raven.capture_exception(e)
  puts "Java exception resulted in this:"
  ap result[:exception]
end

begin
  raises_ruby_exception
rescue Exception => e
  result = Raven.capture_exception(e)
  puts "Ruby exception resulting in this:"
  ap result[:exception]
end
```

Running it for me with sentry-raven 0.13.2, jruby 1.7.19, java build 1.7.0_75-b13, and OSX 10.10.3 results in the following output:

```
I, [2015-05-12T15:30:58.706000 #92349]  INFO -- : ** [Raven] Raven 0.13.2 ready to catch errors
W, [2015-05-12T15:30:58.747000 #92349]  WARN -- : ** [Raven] DEPRECATION WARNING: Calling #send on Raven::Base will be         removed in Raven-Ruby 0.14! Use #send_event instead!
Java exception resulted in this:
{
    :values => [
        [0] {
            :stacktrace => {
                :frames => []
            },
                 :value => "A Java error",
                  :type => "Java::JavaLang::OutOfMemoryError",
                :module => "Java::JavaLang"
        }
    ]
}
W, [2015-05-12T15:31:00.419000 #92349]  WARN -- : ** [Raven] DEPRECATION WARNING: Calling #send on Raven::Base will be         removed in Raven-Ruby 0.14! Use #send_event instead!
Ruby exception resulting in this:
{
    :values => [
        [0] {
            :stacktrace => {
                :frames => [
                    [0] {
                        :project_root => nil,
                         :pre_context => [
                            [0] "end\n",
                            [1] "\n",
                            [2] "begin\n"
                        ],
                              :in_app => false,
                              :lineno => 27,
                            :abs_path => "./raven_bug.rb",
                        :post_context => [
                            [0] "rescue Exception => e\n",
                            [1] "  result = Raven.capture_exception(e)\n",
                            [2] "  puts \"Ruby exception resulting in this:\"\n"
                        ],
                        :context_line => "  raises_ruby_exception\n",
                            :function => "(root)",
                            :filename => "./raven_bug.rb"
                    },
                    [1] {
                        :project_root => nil,
                         :pre_context => [
                            [0] "end\n",
                            [1] "\n",
                            [2] "def raises_ruby_exception\n"
                        ],
                              :in_app => false,
                              :lineno => 15,
                            :abs_path => "./raven_bug.rb",
                        :post_context => [
                            [0] "end\n",
                            [1] "\n",
                            [2] "begin\n"
                        ],
                        :context_line => "  raise RuntimeError.new(\"A Ruby error\")\n",
                            :function => "raises_ruby_exception",
                            :filename => "./raven_bug.rb"
                    }
                ]
            },
                 :value => "A Ruby error",
                  :type => "RuntimeError",
                :module => ""
        }
    ]
}
```

**Comment by dcramer** [comment]
In JRuby is this running as Ruby 1.9 ?

**Comment by dcramer** [comment]
As an aside, what we likely want to do is add a test that specific instruments a java exception (which seems to be what doesnt report here).

**Comment by marshalium** [comment]
This is running as Ruby 1.9. But I get the same results if I run it as Ruby 2.0.

**Comment by marshalium** [comment]
Something else I noticed that is strange is the deprecation warning in the output above:

WARN -- : ** [Raven] DEPRECATION WARNING: Calling #send on Raven::Base will be  removed in Raven-Ruby 0.14! Use #send_event instead!

As you can see in the code, I'm only calling Raven.capture_exception. So it seems like either Raven.capture_exception is deprecated too and the message is wrong, or there is an internal call to #send that should be changed to #send_event.

**Comment by nateberkopec** [comment]
That deprecation warning is from #332. I'll push a fix.

**Comment by dcramer** [comment]
FYI I'm going to, at the very least, get a test to prove this is actually broken (or blame it on Java or something else).

**Comment by dcramer** [comment]
Ah it's because the backtrace lines turn to Java:

org.jruby.Ruby.runScript(Ruby.java:866)
org.jruby.Ruby.runScript(Ruby.java:859)
org.jruby.Ruby.runNormally(Ruby.java:728)
org.jruby.Ruby.runFromMain(Ruby.java:577)
org.jruby.Main.doRunFromMain(Main.java:395)
org.jruby.Main.internalRun(Main.java:290)
org.jruby.Main.run(Main.java:217)
org.jruby.Main.main(Main.java:197)

**Comment by dcramer** [comment]
Fixed in bcf8c28f4895ce977818acded315c9a2f3c1d7bb
